source/common/network/address_impl.cc

```diff
@@ -128,7 +128,10 @@ int InstanceBase::socketFromSocketType(SocketType socketType) const {
   }
 
   int fd = ::socket(domain, flags, 0);
-  RELEASE_ASSERT(fd != -1);
+  if (fd == -1) {
+    throw EnvoyException(
```

Comment:
This is actually a bit scarier than it looks. I hit this recently in https://github.com/envoyproxy/envoy/pull/3230; if you throw exceptions randomly at runtime, code such as gRPC AsyncClient will start following unexpected paths, for example double freeing code.

Reply:
Yeah agree would rather not do this. Can we potentially check this at config load time somehow?